clippy_lints/src/functions/mod.rs

```diff
@@ -354,6 +355,29 @@ declare_clippy_lint! {
     "`impl Trait` is used in the function's parameters"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Triggers when a function with a name ending with `_unchecked` is not marked as unsafe.
+    /// ### Why is this bad?
+    /// `_unchecked` functions may result in memory safety or undefined behaviour, so marking them with unsafe is a way to alert of their uncertainty.
```

Comment:
```suggestion
    /// ### What it does
    /// Triggers when a function with a name ending with `_unchecked` is not marked as unsafe.
    ///
    /// ### Why is this bad?
    /// `_unchecked` functions that have safety requirements should be marked `unsafe`.
    ///
    /// ### Known problems
    /// An `_unchecked` function can be completely safe, with no conditions to maintain memory safety. In those cases they should not be marked `unsafe`.
```

Plus line breaks as needed

Reply:
Fixed